Analysis/Tasks/PWGHF/taskDPlus.cxx

```diff
@@ -23,57 +23,66 @@
+     {"hImpactParameterProngSqSum", "3-prong candidates;squared sum of prong imp. par. (cm^{2});entries", {HistType::kTH1F, {{100, 0., 1.}}}},
+     {"hDecayLengthError", "3-prong candidates;decay length error (cm);entries", {HistType::kTH1F, {{100, 0., 1.}}}},
+     {"hDecayLengthXYError", "3-prong candidates;decay length xy error (cm);entries", {HistType::kTH1F, {{100, 0., 1.}}}},
+     {"hImpactParameterError", "3-prong candidates;impact parameter error (cm);entries", {HistType::kTH1F, {{100, 0., 1.}}}},
+     {"hPtProng0", "3-prong candidates;prong 0 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hPtProng1", "3-prong candidates;prong 1 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hPtProng2", "3-prong candidates;prong 2 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hd0Prong0", "3-prong candidates;prong 0 DCAxy to prim. vertex (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}},
+     {"hd0Prong1", "3-prong candidates;prong 1 DCAxy to prim. vertex (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}},
+     {"hd0Prong2", "3-prong candidates;prong 2 DCAxy to prim. vertex (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}}}};
 
   Configurable<int> d_selectionFlagDPlus{"d_selectionFlagDPlus", 1, "Selection Flag for DPlus"};
+  Configurable<double> cutYCandMax{"cutYCandMax", -1., "max. cand. rapidity"};
 
-  //Filter filterSelectCandidates = (aod::hf_selcandidate::isSelDPlus >= d_selectionFlagDPlus);
+  Filter filterSelectCandidates = (aod::hf_selcandidate_dplus::isSelDplusToPiKPi >= d_selectionFlagDPlus);
 
-  //void process(soa::Filtered<soa::Join<aod::HfCandProng3, aod::HFSelDPlusCandidate>> const& candidates)
-  void process(aod::HfCandProng3 const& candidates)
+  void process(soa::Filtered<soa::Join<aod::HfCandProng3, aod::HFSelDplusToPiKPiCandidate>> const& candidates)
   {
     for (auto& candidate : candidates) {
-      //apply only candidate pre-selections for the moment, not possible in Filter
-      //since expressions do not support binary operators
+      //not possible in Filter since expressions do not support binary operators
       if (!(candidate.hfflag() & 1 << DPlusToPiKPi)) {
         continue;
       }
+      if (cutYCandMax >= 0. && std::abs(YDPlus(candidate)) > cutYCandMax) {
+        continue;
+      }
       registry.fill(HIST("hMass"), InvMassDPlus(candidate));
       registry.fill(HIST("hPt"), candidate.pt());
       registry.fill(HIST("hEta"), candidate.eta());
       registry.fill(HIST("hCt"), CtDPlus(candidate));
       registry.fill(HIST("hDecayLength"), candidate.decayLength());
       registry.fill(HIST("hDecayLengthXY"), candidate.decayLengthXY());
       registry.fill(HIST("hNormalisedDecayLengthXY"), candidate.decayLengthXYNormalised());
+      registry.fill(HIST("hNormalisedDecayLengthXYTimesPOverPt"), candidate.decayLengthXYNormalisedTimesPoverPt());
```

Comment:
Are there any prospects of needing this variable more often?
It does not seem worth creating another column for it if it is just
```c++
candidate.decayLengthXYNormalised() * candidate.P() / candidate.Pt()`
```

Reply:
This variable is not used in analyses at the moment but could be potentially useful. I added it under the assumption that dynamic columns do not have overhead if you do not use them (this is my understanding from what discussed in a past meeting).
What you suggest works only in for loops, right? While we want to be able to use these variables in filters, etc.

Reply:
That is basically what I would like to know. Where has this quantity been used so far in the D+ analyses?

Reply:
This variable is implemented in the AliPhysics Dplus selection code https://github.com/alisw/AliPhysics/blob/0037dc72bea2c498ea03e459a2e8c3db4e3741ad/PWGHF/vertexingHF/AliRDHFCutsDplustoKpipi.cxx#L556 and it is not used in Run2 analyses for the moment. In my opinion, it could potentially be useful in the future and for ML studies (this has to be checked with the Run3 MC productions).

Reply:
If it will be used for ML studies, it will need to be written to a tree and can be calculated then, when the tree is filled.
If it is not expected to be used in the O2 analysis, I don't see a reason for adding it.

Reply:
> If it will be used for ML studies, it will need to be written to a tree and can be calculated then, when the tree is filled.

My understanding is that to fill the tree you will use a table or specify a subset of table columns to write. So. every variable needs to be defined as a dynamic column.
> If it is not expected to be used in the O2 analysis, I don't see a reason for adding it.

It's difficult to say now what would be the final selections of a Run3 analysis. This variable could be useful.

In any case, if this is a blocker for the PR I can remove it.

Reply:
> My understanding is that to fill the tree you will use a table or specify a subset of table columns to write. So. every variable needs to be defined as a dynamic column.

There is no such need in order to fill a tree branch with a variable. Dynamic columns are just functions, so either way, the value is calculated when used.

> It's difficult to say now what would be the final selections of a Run3 analysis. This variable could be useful.
> 
> In any case, if this is a blocker for the PR I can remove it.

From what you have replied, I really don't see any reason to add another column for a simple A * B / C calculation using existing columns, especially when there are no prospects of its usefulness.

Reply:
Ok, removed.